src/test/storage/encoded_string_segment_test.cpp

```diff
@@ -0,0 +1,301 @@
+#include <boost/hana/at_key.hpp>
+
+#include <cctype>
+#include <memory>
+#include <random>
+#include <sstream>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "constant_mappings.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/encoding_type.hpp"
+#include "storage/resolve_encoded_segment_type.hpp"
+#include "storage/segment_encoding_utils.hpp"
+#include "storage/value_segment.hpp"
+
+#include "types.hpp"
+#include "utils/enum_constant.hpp"
+
+namespace opossum {
+
+class EncodedStringSegmentTest : public BaseTestWithParam<SegmentEncodingSpec> {
+ protected:
+  static constexpr auto max_length = 32;
+  static constexpr auto row_count = size_t{1u} << 10;
+
+ protected:
+  std::shared_ptr<ValueSegment<pmr_string>> create_empty_string_value_segment() {
+    auto values = pmr_concurrent_vector<pmr_string>(row_count);
+    return std::make_shared<ValueSegment<pmr_string>>(std::move(values));
+  }
+
+  std::shared_ptr<ValueSegment<pmr_string>> create_empty_string_w_null_value_segment() {
+    auto values = pmr_concurrent_vector<pmr_string>(row_count);
+    auto null_values = pmr_concurrent_vector<bool>(row_count);
+
+    std::default_random_engine engine{};
+    std::bernoulli_distribution bernoulli_dist{0.3};
+
+    for (auto i = 0u; i < row_count; ++i) {
```

Comment:
There are several loops where you use i, but in most you use index. Decide for one and use it consistently.
```suggestion
    for (auto index = size_t{ 0 }; index < row_count; ++index) {
```